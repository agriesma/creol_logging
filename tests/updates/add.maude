load creol-update .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> Configuration [ctor] .
  eq state =
    add(< class("Log", 0) : Class | Version: 0, Inh: noInh, Param: noVid,
          Att: "log" |-> null,
          Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null,
                  Code: skip ; return ( emp ) >,
                < "run" : Method | Param: noVid, Att: "_" |-> null,
                  Code: skip ; return ( emp ) >,
                < "log" : Method | Param: "message", Att: "message" |-> null,
                  "_" |-> null, Code: assign( "log" ; "|-|" ( "log" ::
                  "message" ) ) ; return ( emp ) >,
          Ocnt: 0 >,
      (none))
    
    < class("BankAccount", 0) : Class | Version: 0, Inh: noInh, Param: noVid,
      Att: "bal" |-> null,
      Mtds: < "balance" : Method | Param: noVid, Att: "_" |-> null, "b" |->
              null, "_" |-> null, Code: assign( "b" ; "bal" ) ;
              return ( "b" ) >,
            < "deposit" : Method | Param: "sum", Att: "sum" |-> null, "_" |->
              null, "ccglab0" |-> null, "sum" |-> null, "_" |-> null,
              Code: call( "ccglab0" ; "this" ; "increaseBalance" ; "sum" ) ;
              free( "ccglab0" ) ; return ( emp ) >,
            < "increaseBalance" : Method | Param: "sum", Att: "sum" |-> null,
              "_" |-> null, "sum" |-> null, "_" |-> null,
              Code: assign( "bal" ; "+" ( "bal" :: "sum" ) ) ; return ( emp )
              >,
            < "init" : Method | Param: noVid, Att: "_" |-> null, "_" |->
              null, Code: assign( ( "bal" @ "BankAccount" ) ; int(0) ) ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null, "_" |-> null,
              Code: skip ; return ( emp ) >,
      Ocnt: 0 > .
endm

