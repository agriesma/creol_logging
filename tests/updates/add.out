==========================================
rewrite in PROGRAM : state .
result Configuration: 
< class("BankAccount",0) : Class | Version: 0, Inh: noInh, Param: noVid, Att:
    "bal" |-> null, Mtds:
  < "balance" : Method | Param: noVid, Att: "_" |-> null, "_" |-> null, "b" |->
    null, Code: assign("b" ; "bal") ; return("b") >, < "deposit" : Method |
    Param: "sum", Att: "_" |-> null, "_" |-> null, "ccglab0" |-> null, "sum"
    |-> null, "sum" |-> null, Code: call("ccglab0" ; "this" ; "increaseBalance"
    ; "sum") ; free("ccglab0") ; return(emp) >, < "increaseBalance" : Method |
    Param: "sum", Att: "_" |-> null, "_" |-> null, "sum" |-> null, "sum" |->
    null, Code: assign("bal" ; "+"("bal" :: "sum")) ; return(emp) >, < "init" :
    Method | Param: noVid, Att: "_" |-> null, "_" |-> null, Code: assign("bal"
    @ "BankAccount" ; int(0)) ; return(emp) >, < "run" : Method | Param: noVid,
    Att: "_" |-> null, "_" |-> null, Code: skip ; return(emp) >,
  Ocnt: 0 >
 
< class("Log",1) : Class | Version: 1, Inh: noInh, Param: noVid, Att: "log" |->
    null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(emp)
    >, < "log" : Method | Param: "message", Att: "_" |-> null, "message" |->
    null, Code: assign("log" ; "|-|"("log" :: "message")) ; return(emp) >, <
    "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(emp)
    >,
  Ocnt: 0 >

Bye.
